### PR TITLE
Call Notification and Overlay Dialogs.

### PR DESCRIPTION
**Pull request by peterp**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -2,40 +2,69 @@ import React, { PropTypes } from 'react';
...
+    color: COLORS.BUTTON_ALT_LABEL,
+  },
+  labelDisabled: {
+    color: COLORS.BUTTON_DISABLED_LABEL,
+  },
 });
 
 const Button = ({
-  label,
+  theme = 'orange',
```
Comment (peterp) [inline src/components/Button.js]: We have two alternate styles for the button, orange and white

screen shot 2016-08-12 at 09 25 37

File: src/call/CallNotification.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from 'react';
```
Comment (peterp) [inline src/call/CallNotification.js]: screen shot 2016-08-10 at 18 04 39

File: src/components/CompletedOverlay.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import React, { PropTypes } from 'react';
```
Comment (peterp) [inline src/components/CompletedOverlay.js]: screen shot 2016-08-11 at 15 02 08

File: src/components/LoadingOverlay.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import React, { PropTypes } from 'react';
```
Comment (peterp) [inline src/components/LoadingOverlay.js]: screen shot 2016-08-12 at 09 28 23

File: src/constants/colors.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+export default {
```
Comment (peterp) [inline src/constants/colors.js]: Once I have all the semantic constants in place I'll extract the colours into constants.

File: src/components/Button.js [hunk 6]
```diff
@@ -2,40 +2,69 @@ import React, { PropTypes } from 'react';
...
+      style={[
+        styles.button,
+        styles[`button_${theme}`],
+        disabled && styles.buttonDisabled,
+      ]}
+    >
+      <Text
+        style={[
+          styles.label,
+          styles[`label_${theme}`],
```
Comment (justinvdm) [inline src/components/Button.js]: As you suggested discussed on slack, should we use a constant for configuring themes here instead?

File: src/components/SpinningImage.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      outputRange: ['0deg', '360deg'],
+    });
+    return (
+      <Animated.Image
+        source={IMAGE_LOADER}
+        style={{
+          width: 80,
+          height: 80,
+          transform: [{ rotate: spin }],
+        }}
```
Comment (justinvdm) [inline src/components/SpinningImage.js]: Would it be possible to add this to the stylesheet at the top?

File: src/components/SpinningImage.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    this.spin();
+  }
+
+  spin() {
+    this.state.spinValue.setValue(0);
+    Animated.timing(
+      this.state.spinValue,
+      {
+        toValue: 1,
+        duration: 2000,
```
Comment (justinvdm) [inline src/components/SpinningImage.js]: Would be cool if we could use constants here.

**Comment by justinvdm** [comment]
Left some comments, looks good :) I think we should note down somewhere that these components still need tests (so we can do that in a separate PR and not hold up this one).

Really like the generic components and the use of them from non-generic components! :)

Are the overlays going to be used by something outside of this PR?

**Pull request merged**